### PR TITLE
Updated summary, especially parts about UKF and BF.

### PR DESCRIPTION
**Pull request by TheKastenkarl**

### CODE REVIEW

File: robot_motion_planning_summary.tex [hunk 1]
```diff
@@ -183,213 +494,364 @@ \subsubsection{Weighted Randomized Tree Expansion}
...
-\subsection{Path and Map Smoothing}
-The paths obtained by random sampling techniques are not necessarily the most efficient paths. Path smoothing techniques are used to optimize the path for length, energy, distance to obstacles etc. 
+\section{Probabilistic Robotics}
+\subsection{Kalman Filter}
+The basic idea of a Kalman filter is that you interpret your robots state as a normal (gaussian) distribution with mean and variance.
+Besides the estimate for the state $x$ you also have some estimate for the uncertainty in your value for $x$ that is represented through a normal distribution and its variance / covariance matrix $P$.
+You also model your sensors with white noise (gaussian, zero-mean).\\
 
-To grow the distance to obstacles, the map can be smoothed by itself before planning. For example, a 2D-convolution with a Gaussian filter kernel leads to a non-zero probability distribution around the edges of the obstacles (edge smoothing).
+\textit{\textbf{Explanantion from the internet:}}
```
Comment (MichaelGrupp) [inline robot_motion_planning_summary.tex]: This heading looks out of place, you might as well remove it. ("from the internet" is not a source)

File: robot_motion_planning_summary.tex [hunk 2]
```diff
@@ -31,147 +32,457 @@ \section{Introduction}
...
+\textbf{Configuration space (\textit{C-space}):} The space of all possible configurations, that a robot can have in a certain workspace. \\
+\textbf{Dimension of the C-space/DoF:} Minimum number of parameters needed to specify the configuration. Because there can also be angles in the topology of the C-space, it is usually not a Cartesian space (e.g. $2\pi = 4\pi$, cyclic repetition along angle axis). \\
+\textbf{Common robot models:} Single points, single rigid bodies and multiple rigid bodies with joints. \\
+\textbf{Non-holonomic robot:} Robot, that cannot move freely on the
+configuration space manifold because it is constrained in some way. For example, a car cannot drive sideways. \\
+\textbf{Path:} Continuous curve in configuration space $C$ connecting two configurations $q$ and $q'$:
+$\tau: s \in [0,1] \rightarrow \tau(s) \in C$ s.t. $\tau(0) = q$ and $\tau(1) = q'$\\
+\textbf{Trajectory:} A trajectory is a path parametrized by time:
+$\tau: t \in [0, T] \rightarrow \tau(t) \in C$.
+Possible constraints: Smoothness, minimum length, minimum time, ...
```
Comment (MichaelGrupp) [inline robot_motion_planning_summary.tex]: No capitalization after the colon here. Please also check the other places in the text.
https://www.grammarbook.com/blog/commas/capitalization-with-colons/

File: robot_motion_planning_summary.tex (hunk 1 above)
Comment (TheKastenkarl) [reply robot_motion_planning_summary.tex]: Good point, combined the two paragraphs.

File: robot_motion_planning_summary.tex (hunk 2 above)
Comment (TheKastenkarl) [reply robot_motion_planning_summary.tex]: Changed it everywhere in the text where I could find it.

File: robot_motion_planning_summary.tex [hunk 3]
```diff
@@ -31,147 +32,457 @@ \section{Introduction}
...
+\begin{itemize}
+	\item \textbf{Parametrization of orientations by rotation matrix:}\\
+		$R=
+		\begin{psmallmatrix}
+			r_{11} & r_{12} & r_{13}\\
+			r_{21} & r_{22} & r_{23}\\
+			r_{31} & r_{32} & r_{33}
+		\end{psmallmatrix}$
+		with $\det(R)=1$ and $R^T R=I$ (or $R R^T=I$) \\
+		Columns of $R$: coordinate axes of new coordinate System
```
Comment (MichaelGrupp) [inline robot_motion_planning_summary.tex]: Could be added as advantage: rotation matrices can be embedded into a 4x4 SE(3) transformation matrix together with the position. Concatenation of transformations via matrix multiplication (same for pure rotations).

Drawback: over-parameterized

File: robot_motion_planning_summary.tex [hunk 4]
```diff
@@ -31,147 +32,457 @@ \section{Introduction}
...
+			0 & 1 & 0\\
+			-s\beta & 0 & c\beta
+		\end{psmallmatrix}
+		\begin{psmallmatrix}
+			1 & 0 & 0\\
+			0 & c\alpha & -s\alpha\\
+			0 & s\alpha & c\alpha
+		\end{psmallmatrix}
+	\end{align*}
+	Problem: singularities
```
Comment (MichaelGrupp) [inline robot_motion_planning_summary.tex]: The many possible sequence conventions are also a drawback, easy to get wrong. And they are harder to concatenate.

File: robot_motion_planning_summary.tex [hunk 5]
```diff
@@ -31,147 +32,457 @@ \section{Introduction}
...
+	– $n =$ number of joints\\
+	– $f_i =$ DoF of the $i^{th}$ joint
+\end{itemize}
 
-A configuration describes the state of a robot as a combination of different configurations of its degrees of freedom. For example, a mobile robot moves in directions $x,y$ and can turn its yaw heading $\theta$ - a configuration of this robot could be $c = (1,3,30^\circ)^\mathrm{T}$ The configuration space (\textit{C-space}) is the space of all possible configurations, that a robot can have in a certain workspace.
-
-The dimension of the C-space is equal to the degrees of freedom of the robot (minimum number of parameters needed to specify the configuration). Because there can also be angles in the topology of the C-space, it is usually not a cartesian space (e.g. $2\pi = 4\pi$, cyclic repetition along angle axis).
+\subsection{Parametrizations of configurations}
+Configuration: $\mathbold{q}=(position, orientation)=(x,y,z,\cdot)$.
+There exist different ways how to parametrize orientations:
```
Comment (MichaelGrupp) [inline robot_motion_planning_summary.tex:77]: Angle-axis parameterization could be also added, I don't know if this is relevant for this lecture.

File: robot_motion_planning_summary.tex [hunk 6]
```diff
@@ -31,147 +32,457 @@ \section{Introduction}
...
+			1 & 0 & 0\\
+			0 & c\alpha & -s\alpha\\
+			0 & s\alpha & c\alpha
+		\end{psmallmatrix}
+	\end{align*}
+	Problem: singularities
+	\item \textbf{Parametrization of orientations by unit quaternions:}\\
+	$\mathbold{u} = (\cos(\theta/2), n_x \sin(\theta/2), n_y \sin(\theta/2), n_z sin(\theta/2))$\\
+	with $\lVert \mathbold{n} \rVert = 1$ ($\mathbold{n}$: rotation vector, $\theta$: rotation angle)\\
+	Advantages: compact, no singularities, natural orientation representation
```
Comment (MichaelGrupp) [inline robot_motion_planning_summary.tex]: Could be also added (not sure if this is already in "compact"): efficient computations (multiplication, inverse etc)

**Review by MichaelGrupp — APPROVED** [review]
Looks good! I added a few comments on the rotation part, feel free to add this if you want.

File: robot_motion_planning_summary.tex (hunk 6 above)
Comment (TheKastenkarl) [reply robot_motion_planning_summary.tex]: I added it.

File: robot_motion_planning_summary.tex (hunk 5 above)
Comment (TheKastenkarl) [reply robot_motion_planning_summary.tex:77]: I added it.

File: robot_motion_planning_summary.tex (hunk 4 above)
Comment (TheKastenkarl) [reply robot_motion_planning_summary.tex]: I added it.

File: robot_motion_planning_summary.tex (hunk 3 above)
Comment (TheKastenkarl) [reply robot_motion_planning_summary.tex]: I added it.

**Comment by TheKastenkarl** [comment]
Good suggestions, thanks!

**Pull request merged**